PE-D Bug fixing

SAKICHANNN opened the pull request.

Commit:
Fix #172

Commit:
Fix #162

Commit:
Fix #172

Commit:
Fix #152

Commit:
Fix #142

Commit:
Update User Guide: document name special characters and duplicate phone number constraint

Commit:
Fix #134: Replace Address Book with CampusBook and use "contact list" as synonym for generic address book

Commit:
Fix code coverage: add test for single error case in ImportCommand

DJsudartha commented on docs/UserGuide.md:
```diff
@@ -489,11 +499,11 @@ Furthermore, certain edits can cause the CampusBook to behave in unexpected ways
-**Q**: Why does the applicaiton allow same names with different capitalisation (e.g. John & john) to be added?<br>
-**A**: This is to prevent overzealous input validation, as there are some names with the same characters, but different capitalisation (e.g. McKenzie & Mckenzie).<br>
+**Q**: Can I add contacts with special characters in their names (e.g. O'Connor, José)?<br>
+**A**: Yes! The application supports names with special characters such as apostrophes (`'`), hyphens (`-`), slashes (`/`), periods (`.`), and accented characters (e.g. é, ó). This allows for realistic names like `O'Connor`, `Ravi Kumar s/o Suresh Kumar`, and `José`.<br>
```
Maybe note down that special characters shouldn't include any of the prefixes we mentioned

Commit:
Fix duplicate contact definition: prevent same name OR same phone number

- Update Person.isSamePerson() to check both name and phone number
- Update UserGuide to clarify duplicate contact rules
- Contacts with same name OR same phone number are considered duplicates
- Prevents both duplicate names and duplicate phone numbers

Commit:
Update UserGuide: add warning about command prefixes in names

- Add warning in name field notes about not using command prefixes
- Add note in FAQ section about command prefix limitation
- Clarifies that prefixes like n/, p/, e/ in names will cause parsing errors
- Addresses code review comments about prefix handling

Commit:
Update error messages to reflect duplicate detection logic

- Update AddCommand error message to specify 'same name or phone number'
- Update EditCommand error message to specify 'same name or phone number'
- Update DuplicatePersonException javadoc to clarify duplicate criteria
- Addresses code review comment about ensuring error messages are updated

Commit:
Add tests for names with command prefix patterns

- Add test cases in NameTest to verify prefix-like patterns without space are valid at Name level
- Add test case in AddCommandParserTest to verify names with prefix (with space) cause parsing errors
- Documents that while names with prefix patterns are valid at validation level, they cause issues when used in commands
- Addresses code review comment about verifying prefix behavior in names

restinnotes committed:
Merge branch 'master' into branch-ped-fixing

Commit:
Fix checkstyle: break long lines in AddCommand and EditCommand

- Fix line length violations in MESSAGE_DUPLICATE_PERSON constants
- Split long lines to comply with 120 character limit

Commit:
Fix checkstyle: remove trailing whitespace in AddCommandParserTest

Commit:
Unify singular/plural forms: use (s) notation for variable quantities

- Remove conditional logic for singular/plural forms in code
- Use (s) notation consistently for quantities that may vary (0, 1, or many)
- Keep fixed singular/plural forms where quantity is certain (e.g., 'Listed all persons')
- Update all test expectations to match new message formats
- Update DeveloperGuide.md test case descriptions

Changes:
- ImportCommand: contact(s), duplicate row(s), invalid row(s)
- Messages: person(s) listed, person(s) deleted, person(s) found, field(s)
- DeleteCommand: person(s) for batch delete, removed conditional logic
- FindCommand: removed conditional, unified to person(s) listed
- JsonSerializableAddressBook: duplicate person(s) (variable quantity)
- ListCommand: kept 'persons' (certain plural with 'all')
- Updated all related test files to match new formats

Commit:
Fix UserGuide: add missing line break after 'How Fields Work' heading

Commit:
Restore Q&A about allowing names with different capitalisation

- Add back FAQ explaining why same names with different capitalisation are allowed
- This aligns with the code behavior where names are case-sensitive
- Examples: John & john, McKenzie & Mckenzie can be added if phone numbers differ

DJsudartha committed:
Merge branch 'master' into branch-ped-fixing
